prov/sockets/src/sock_cq.c

```diff
@@ -406,7 +406,7 @@ static ssize_t sock_cq_sreadfrom(struct fid_cq *cq, void *buf, size_t count,
 		} while (ret > 0);
 	}
 
-	return (ret == 0 || ret == -FI_ETIMEDOUT) ? -FI_EAGAIN : ret;
+	return (ret == 0 || ret == -FI_ETIMEDOUT || ret == -EINTR) ? -FI_EAGAIN : ret;
```

Comment:
There was some discussion either around this or around a similar change way, way back.  @sydidelot may remember the details better that I can.

My general view is that we should not hide EINTR or other expected failures from the app.  Is there a reason to hide this?

Reply:
Most of the client code I've seen doesn't check for -EINTR but only checks for -EAGAIN.
  OpenMPI: https://github.com/open-mpi/ompi/blob/master/ompi/mca/mtl/ofi/mtl_ofi.h#L227
  pingpong.c: https://github.com/ofiwg/libfabric/blob/master/util/pingpong.c#L1088
I can't remember how I encountered this exactly...

Reply:
It's possible that I witnessed this when I suspended and resumed the process (perhaps ctrl-Z, fg; or maybe it was from attaching gdb) and it aborted with -EINTR. Again, my memory is fuzzy, but I think it was in this code path: https://github.com/open-mpi/ompi/blob/89c1b0c7edb173a2c94a3641767d0c86c0ad9021/orte/orted/orted_comm.c#L218

Reply:
I can believe that apps aren't checking for anything other than EAGAIN.  But the question is whether it's the responsibility of libfabric to convert other error codes to EAGAIN, or for the app to check other values.  Basically, what error codes _should_ OFI hide from the app?

Reply:
I posted a message to the ofiwg to gather community input, if any, on this PR.